services/softlayer_virtual_guest.go

```diff
@@ -1200,6 +1200,21 @@ func (slvgs *softLayer_Virtual_Guest_Service) GetLocalDiskFlag(instanceId int) (
 	return false, errors.New(fmt.Sprintf("Failed to check the disk type (local or SAN) of that virtual guest with id '%d', got '%s' as response from the API.", instanceId, res))
 }
 
+func (slvgs *softLayer_Virtual_Guest_Service) GetBlockDevices(instanceId int) ([]datatypes.SoftLayer_Virtual_Guest_Block_Device, error) {
+	response, _, err := slvgs.client.GetHttpClient().DoRawHttpRequest(fmt.Sprintf("%s/%d/getBlockDevices.json", slvgs.GetName(), instanceId), "GET", new(bytes.Buffer))
```

Comment:
Due to [PR 136](https://github.com/maximilien/softlayer-go/pull/136), http return code has already been checked in [http_client.go](https://github.com/bondyk/softlayer-go/blob/abaec9812b2470f1f13b88eb06714fc280b88c88/client/http_client.go#L210), so if the http return code > 400, http client will return an error, there is no need to check http code in `GetBlockDevices` and all alike functions.